src/metabase/lib/temporal_bucket.cljc

```diff
@@ -102,19 +101,19 @@
    unit :- [:maybe :keyword]]
   (let [n    (interval-n->int n)
         unit (or unit :day)]
-    (cond
-      (zero? n)
-      (i18n/tru "Now")
-
-      (neg? n)
-      ;; this should legitimately be lowercasing in the user locale. I know system locale isn't necessarily the same
-      ;; thing, but it might be. This will have to do until we have some sort of user-locale lower-case functionality
-      #_{:clj-kondo/ignore [:discouraged-var]}
-      (i18n/tru "{0} {1} ago" (abs n) (str/lower-case (describe-temporal-unit (abs n) unit)))
-
-      :else
-      #_{:clj-kondo/ignore [:discouraged-var]}
-      (i18n/tru "{0} {1} from now" n (str/lower-case (describe-temporal-unit n unit))))))
+    (case (keyword unit)
+      :default     (lib.temporal-bucket.util/relative-datetime-tru n "default period")
+      :millisecond (lib.temporal-bucket.util/relative-datetime-tru n "millisecond")
+      :second      (lib.temporal-bucket.util/relative-datetime-tru n "second")
+      :minute      (lib.temporal-bucket.util/relative-datetime-tru n "minute")
+      :hour        (lib.temporal-bucket.util/relative-datetime-tru n "hour")
+      :day         (lib.temporal-bucket.util/relative-datetime-tru n "day")
+      :week        (lib.temporal-bucket.util/relative-datetime-tru n "week")
+      :month       (lib.temporal-bucket.util/relative-datetime-tru n "month")
+      :quarter     (lib.temporal-bucket.util/relative-datetime-tru n "quarter")
+      :year        (lib.temporal-bucket.util/relative-datetime-tru n "year")
+      ;; else
+      (lib.temporal-bucket.util/relative-datetime-tru n "unknown unit"))))
```

Comment:
Looks like the `case` is only needed to convert the keyword to some text (which in the majority of cases could be done by calling `name` on it).